Source/Evolve/WarpXEvolve.cpp

```diff
@@ -370,20 +370,35 @@ WarpX::Evolve (int numsteps)
         }
 
         exit_loop_due_to_interrupt_signal = SignalHandling::TestAndResetActionRequestFlag(SignalHandling::SIGNAL_REQUESTS_BREAK);
-        if (cur_time >= stop_time - 1.e-3*dt[0] || exit_loop_due_to_interrupt_signal) {
+        if (exit_loop_due_to_interrupt_signal) {
```

Comment:
This change makes sense to me.  The fudge factor is 7 years old or older, do you know why it was there?

Reply:
I think there was concern about what happens if `cur_time` lands right near `stop_time`, for instance if a user had set `stop_time = 100*dt`. I don't think this would be an issue, and if so, the user should set `max_step = 100`.
Note that the check of `cur_time` here is unnecessary since it is checked at the start of the loop anyway.

Reply:
I did see that.  It makes sense to me to take out this seemingly unnecessary magic number and logic